src/client/pool.rs

```diff
@@ -42,6 +43,8 @@ pub(super) trait Poolable: Unpin + Send + Sized + 'static {
     fn can_share(&self) -> bool;
 }
 
+pub trait Key: Eq + Hash + Clone + Debug + Unpin {}
```

Comment:
Do we need the separate trait?

It looks like this may exist just to reduce the list at `K: Eq + Hash + ...`. But if the trait doesn't provide anything new, it would provide a hurdle to users: they'd need to `impl Key for MyThing`. That'd be more annoying if `MyThing` were from a different crate, so they _couldn't_ implement it.

Reply:
That's na great point, new to `Rust` things!!

Reply:
@seanmonstar we could do a blanket implementation? Or do you prefer putting all the individual traits on each bounds?